akd/src/directory.rs

```diff
@@ -275,13 +307,60 @@ impl<S: Storage + Sync + Send, V: AkdVRF> Directory<S, V> {
         }
     }
 
+    /// Poll for changes in the epoch number of the AZKS struct
+    /// stored in the storage layer. If an epoch change is detected,
+    /// the object cache (if present) is flushed immediately so
+    /// that new objects are retrieved from the storage layer against
+    /// the "latest" epoch. There is a "special" flow in the storage layer
+    /// to do a storage-layer retrieval which ignores the cache
+    pub async fn poll_for_azks_changes(
+        &self,
+        period: tokio::time::Duration,
+        change_detected: Option<tokio::sync::mpsc::Sender<()>>,
+    ) -> Result<(), AkdError> {
+        // Retrieve the same AZKS that all the other calls see (i.e. the version that could be cached
+        // at this point). We'll compare this via an uncached call when a change is notified
+        let mut last = Directory::<S, V>::get_azks_from_storage(&self.storage, false).await?;
+
+        loop {
+            // loop forever polling for changes
+            tokio::time::sleep(period).await;
+
+            let latest = Directory::<S, V>::get_azks_from_storage(&self.storage, true).await?;
```

Comment:
Maybe to ignore/log errors, or retry couple of times here and below instead of crashing (if transient error)?

Reply:
Bubbling up the error just means the caller of this function needs to handle it. We could handle the error handling internally, but there might be non-transient errors which we want passed up. It just makes the inner handling logic more complicated.

If it were to fail, the caller can catch, log/print/etc whatever they want, and re-call the function to continue polling.